Error occurred while enabling TownyResources v0.1.0 (Is it up to date?)

I got a new error :( the resource offer categories are now loading.. 

[16:09:39 INFO]: [TownyResources] Towny version 0.97.1.0 found.
[16:09:39 INFO]: [TownyResources] SiegeWar Integration Not Enabled
[16:09:39 INFO]: [TownyResources] DynmapTowny Integration Enabled
[16:09:39 INFO]: [TownyResources] Slimefun Integration Not Enabled
[16:09:39 INFO]: [TownyResources] LanguageUtils Integration Not Enabled
[16:09:39 INFO]: [TownyResources] Config Loaded
[16:09:39 INFO]: [TownyResources] Lang: english.yml v0.06 loaded.
[16:09:39 INFO]: [TownyResources] All Resource Offer Categories Loaded
[16:09:39 ERROR]: Error occurred while enabling TownyResources v0.1.0 (Is it up to date?)
java.lang.NoClassDefFoundError: net/md_5/bungee/api/chat/BaseComponent
	at io.github.townyadvanced.townyresources.TownyResources.loadAll(TownyResources.java:68) ~[?:?]
	at io.github.townyadvanced.townyresources.TownyResources.onEnable(TownyResources.java:34) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:343) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:479) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:480) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:406) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:426) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:205) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:787) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$0(MinecraftServer.java:155) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_301]
Caused by: java.lang.ClassNotFoundException: net.md_5.bungee.api.chat.BaseComponent
	at java.net.URLClassLoader.findClass(Unknown Source) ~[?:1.8.0_301]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:167) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:96) ~[craftbukkit-1.16.5.jar:2991-Bukkit-f3f3094]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_301]
	at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_301]
	... 12 more

--- a/pom.xml
+++ b/pom.xml
@@ -2,7 +2,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.github.townyadvanced</groupId>
   <artifactId>TownyResources</artifactId>
-  <version>0.1.0</version>
+  <version>0.1.1</version>
   <name>townyresources</name> <!-- Leave lower-cased -->
 
   <properties>

--- a/src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java
+++ b/src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java
@@ -14,7 +14,6 @@ import io.github.townyadvanced.townyresources.settings.TownyResourcesTranslation
 import io.github.townyadvanced.townyresources.util.ActionBarUtil;
 import io.github.townyadvanced.townyresources.util.TownyResourcesMessagingUtil;
 import org.bukkit.Bukkit;
-import org.bukkit.ChatColor;
 import org.bukkit.DyeColor;
 import org.bukkit.Material;
 import org.bukkit.entity.*;
@@ -397,7 +396,7 @@ public class PlayerExtractionLimitsController {
             String errorString = TownyResourcesTranslation.of("msg_error_daily_extraction_limit_reached", translatedCategoryName, categoryExtractionLimit);
             //Send temporary action bar message
             try {
-                ActionBarUtil.sendActionBarMessage(player, errorString);
+                ActionBarUtil.sendActionBarErrorMessage(player, errorString);
             } catch (Exception ignored) {
                 //There may be an exception on older servers which cannot load the md5 action bar classes
             }

--- a/src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java
+++ b/src/main/java/io/github/townyadvanced/townyresources/controllers/PlayerExtractionLimitsController.java
@@ -11,9 +11,8 @@ import io.github.townyadvanced.townyresources.objects.CategoryExtractionRecord;
 import io.github.townyadvanced.townyresources.objects.ResourceExtractionCategory;
 import io.github.townyadvanced.townyresources.settings.TownyResourcesSettings;
 import io.github.townyadvanced.townyresources.settings.TownyResourcesTranslation;
+import io.github.townyadvanced.townyresources.util.ActionBarUtil;
 import io.github.townyadvanced.townyresources.util.TownyResourcesMessagingUtil;
-import net.md_5.bungee.api.ChatMessageType;
-import net.md_5.bungee.api.chat.TextComponent;
 import org.bukkit.Bukkit;
 import org.bukkit.ChatColor;
 import org.bukkit.DyeColor;
@@ -397,7 +396,11 @@ public class PlayerExtractionLimitsController {
             int categoryExtractionLimit = categoryExtractionRecord.getResourceExtractionCategory().getCategoryExtractionLimitItems();
             String errorString = TownyResourcesTranslation.of("msg_error_daily_extraction_limit_reached", translatedCategoryName, categoryExtractionLimit);
             //Send temporary action bar message
-            player.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent(ChatColor.DARK_RED + errorString));
+            try {
+                ActionBarUtil.sendActionBarMessage(player, errorString);
+            } catch (Exception ignored) {
+                //There may be an exception on older servers which cannot load the md5 action bar classes
+            }
             //Send longer-lasting chat message
             TownyResourcesMessagingUtil.sendErrorMsg(player, errorString);                    
             categoryExtractionRecord.setNextLimitWarningTime(System.currentTimeMillis() + cooldownAfterDailyLimitWarningMessageMillis);

--- a/src/main/java/io/github/townyadvanced/townyresources/util/ActionBarUtil.java
+++ b/src/main/java/io/github/townyadvanced/townyresources/util/ActionBarUtil.java
@@ -1,0 +1,17 @@
+package io.github.townyadvanced.townyresources.util;
+
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.TextComponent;
+import org.bukkit.ChatColor;
+import org.bukkit.entity.Player;
+
+/**
+ * FYI if the MC version is not high enough this class may fail to load with a "java.lang.noClassDefFound" error
+ * But its ok because TownyResources always sends a text window message anyway
+ */
+public class ActionBarUtil {
+
+    public static void sendActionBarMessage(Player player, String text) {
+        player.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent(ChatColor.DARK_RED + text));
+    }
+}

--- a/src/main/java/io/github/townyadvanced/townyresources/util/ActionBarUtil.java
+++ b/src/main/java/io/github/townyadvanced/townyresources/util/ActionBarUtil.java
@@ -11,7 +11,7 @@ import org.bukkit.entity.Player;
  */
 public class ActionBarUtil {
 
-    public static void sendActionBarMessage(Player player, String text) {
+    public static void sendActionBarErrorMessage(Player player, String text) {
         player.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent(ChatColor.DARK_RED + text));
     }
 }

